src/solvers/radau5.py

```diff
@@ -134,9 +151,12 @@ def f(t, y):
                 ret = 0
                 try:
                     rhs = self.problem.rhs(t, y)
-                except(N.linalg.LinAlgError,ZeroDivisionError,AssimuloRecoverableError):
+                except BaseException as E:
                     rhs = y.copy()
-                    ret = -1 #Recoverable error
+                    if isinstance(E, (N.linalg.LinAlgError, ZeroDivisionError, AssimuloRecoverableError)): ## recoverable
+                        ret = -1 #Recoverable error
+                    else:
+                        ret = -2 #Non-recoverable
```

Comment:
Good but it seems the 4 last lines are repeat the same everywhere we can have BaseException thrown.  Maybe this could be one single method so the return code are only specified once.

Reply:
This could be done and the best place for this would be in a function in /lib/radau_core.py:Radau_Comon, which is the shared parent class for all classes in radau5.py. 
While this does save a few lines of code, I think it makes it less readable. Thoughts?